core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -85,6 +85,9 @@ private[spark] class Executor(
     env.blockManager.initialize(conf.getAppId)
   }
 
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
sorry I forgot about this in my last round.  I don't understand the first sentence of your reply -- why can't you get the hostname & port?  it looks like you can do exactly what you suggested to get the host & port from the rpc env with `executorMetrics.setHostPort(env.rpcEnv.address.hostPort)`